config/ontoemma_nn_all.json

```diff
@@ -0,0 +1,107 @@
+{
+  "dataset_reader": {
+    "type": "ontology_matcher",
+    "name_token_indexers": {
+      "tokens": {
+        "type": "single_id",
+        "lowercase_tokens": true
+      },
+      "token_characters": {
+        "type": "characters"
+      }
+    },
+    "token_only_indexer": {
+      "tokens": {
+        "type": "single_id",
+        "lowercase_tokens": true
+      }
+    }
+  },
+  "train_data_path": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/training/ontoemma.context.train",
+  "validation_data_path": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/training/ontoemma.context.dev",
+  "model": {
+    "type": "ontoemmaNN",
+    "name_text_field_embedder": {
+            "tokens": {
+                    "type": "embedding",
+                    "pretrained_file": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/embeddings/weights100.txt.gz",
+                    "embedding_dim": 100,
+                    "trainable": false
+            },
+            "token_characters": {
+                    "type": "character_encoding",
+                    "embedding": {
+                      "embedding_dim": 8
+                    },
+                    "encoder": {
+                      "type": "cnn",
+                      "embedding_dim": 8,
+                      "num_filters": 50,
+                      "ngram_filter_sizes": [5]
+                    },
+                    "dropout": 0.2
+            }
+    },
+    "context_text_field_embedder": {
+            "tokens": {
+                    "type": "embedding",
+                    "pretrained_file": "/net/nfs.corp/s2-research/scigraph/data/ontoemma/embeddings/weights100.txt.gz",
+                    "embedding_dim": 100,
+                    "trainable": false
```

Comment:
If you aren't training the embeddings, you can probably afford to make the LSTMs a little larger, as you have quite a bit of data (if the bottleneck here is speed, ignore this). Also, did you resolve that CUDNN issue? If not, we should open an issue in Pytorch. Let me know if you want me to dig into that more, particularly if it is the only thing stopping you increasing the batch size to train this thing faster.

Reply:
I still can't increase the batch size much :/ I think the way TimeDistributed unfurls the data still causes the cudnn contiguous error, which is unfortunate since I'm only using about half the memory on the GPU. Let me know if you have any thoughts on how I could improve this.